bump-up.command

```diff
@@ -19,6 +19,30 @@
 # under the License.
 #
 
+MODULES=("metrics-common" "metadata-store-directory-common" "zookeeper-api" "helix-common" "helix-core" \
+"helix-admin-webapp" "helix-front" "helix-rest" "helix-lock" "helix-view-aggregator" "helix-agent")
+
+update_pom_version() {
+  pom=$1
+  echo "bump up $pom"
+  sed -i "s/${version}/${new_version}/g" $pom
```

Comment:
Thanks for refactoring! Another nagging issue we've had with this script is that if `sed` isn't able to locate the correct `version`, it just fails out, and those who aren't familiar with the way bump-up.command would get stuck. 

I think it might be helpful to add some error log based on the exit code - any thoughts?

Reply:
Hey @narendly , thanks for the comment. 
Not sure if I understand it correctly, but it seems `sed` doesn't complain if the pattern is not found. We actually need to do extra work to make it error out in such case https://unix.stackexchange.com/questions/55810/is-there-an-option-to-make-sed-fail-if-pattern-not-found

Reply:
Added a check with `grep` instead since it's already there and is a smaller change.

Reply:
For Mac, sed may not work.

Reply:
Thanks @junkaixue  I believe this is existing behavior though.
I updated the command to make it work on both Mac and linux, referring to https://stackoverflow.com/questions/4247068/sed-command-with-i-option-failing-on-mac-but-works-on-linux